core/src/test/java/tech/pegasys/eth2signer/core/multikey/DirectoryBackedArtifactSignerProviderTest.java

```diff
@@ -0,0 +1,209 @@
+import tech.pegasys.eth2signer.core.multikey.metadata.SigningMetadataException;
+import tech.pegasys.eth2signer.core.signing.ArtifactSigner;
+import tech.pegasys.eth2signer.crypto.KeyPair;
+import tech.pegasys.eth2signer.crypto.SecretKey;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.Collection;
+import java.util.Optional;
+
+import org.apache.tuweni.bytes.Bytes;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.api.io.TempDir;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+class DirectoryBackedArtifactSignerProviderTest {
+  @TempDir Path configsDirectory;
+  @Mock private SignerParser signerParser;
+
+  private static final String FILE_EXTENSION = ".yaml";
+  private static final String PUBLIC_KEY =
+      "989d34725a2bfc3f15105f3f5fc8741f436c25ee1ee4f948e425d6bcb8c56bce6e06c269635b7e985a7ffa639e2409bf";
+  private static final String PRIVATE_KEY =
+      "000000000000000000000000000000003ee2224386c82ffea477e2adf28a2929f5c349165a4196158c7f3a2ecca40f35";
+
+  private ArtifactSigner artifactSigner;
+  private DirectoryBackedArtifactSignerProvider signerProvider;
+
+  @BeforeEach
+  void setup() {
+    signerProvider = new DirectoryBackedArtifactSignerProvider(configsDirectory, signerParser);
+    artifactSigner = createArtifactSigner(PRIVATE_KEY);
+  }
+
+  @Test
+  void signerReturnedForValidMetadataFile() throws IOException {
+    final String filename = PUBLIC_KEY;
+    createFile(filename);
+    when(signerParser.parse(any())).thenReturn(artifactSigner);
+
+    final Optional<ArtifactSigner> signer = signerProvider.getSigner(PUBLIC_KEY);
+    assertThat(signer).isNotEmpty();
+    assertThat(signer.get().getIdentifier()).isEqualTo("0x" + PUBLIC_KEY);
+    verify(signerParser).parse(pathEndsWith(filename));
+  }
+
+  @Test
+  void signerReturnedWhenIdentifierHasCaseMismatchToFilename() throws IOException {
+    final String filename = PUBLIC_KEY.toUpperCase();
+    createFile(filename);
+    when(signerParser.parse(any())).thenReturn(artifactSigner);
+
+    final Optional<ArtifactSigner> signer = signerProvider.getSigner(PUBLIC_KEY);
+    assertThat(signer).isNotEmpty();
+    assertThat(signer.get().getIdentifier()).isEqualTo("0x" + PUBLIC_KEY);
+    verify(signerParser).parse(pathEndsWith(filename));
+  }
+
+  @Test
+  void signerReturnedWhenHasHexPrefix() throws IOException {
+    final String metadataFilename = PUBLIC_KEY;
+    createFile(metadataFilename);
+    when(signerParser.parse(any())).thenReturn(artifactSigner);
+
+    final Optional<ArtifactSigner> signer = signerProvider.getSigner("0x" + PUBLIC_KEY);
+
+    assertThat(signer).isNotEmpty();
+    verify(signerParser).parse(pathEndsWith(metadataFilename));
+  }
+
+  @Test
+  void signerReturnedWhenHasUpperCaseHexPrefix() throws IOException {
```

Comment:
nit: is this really a valid test? I.e. hex is deliberately 0x (not 0X), I thought ...

Reply:
It should be, but I was just thinking about case insensitive filesystems where it might be returned as 0X being part of the filename